lib/OpSendMsg.h

```diff
@@ -59,6 +60,13 @@ struct OpSendMsg {
         if (sendCallback_) {
             sendCallback_(result, messageId);
         }
+        for (const auto& trackerCallback : trackerCallbacks_) {
+            trackerCallback(result);
+        }
+    }
+
+    void addTrackerCallback(std::function<void(Result)> trackerCallback) {
```

Comment:
nit
```suggestion
    void addTrackerCallback(const std::function<void(Result)> trackerCallback) {
```

Reply:
It's common to pass the `std::function` by value. Regarding the value argument, adding `const` does not harm but it also does not bring any benefit.

Take https://en.cppreference.com/w/cpp/string/basic_string/basic_string for example, the constructors that accept integral types don't add `const` qualifiers.

```c++
basic_string( size_type count, CharT ch,
              const Allocator& alloc = Allocator() );
```